client/go/cmd/prod.go

```diff
@@ -377,18 +374,29 @@ func prompt(r *bufio.Reader, question, defaultAnswer string, validator func(inpu
 	return input, nil
 }
 
-func verifyTests(testsParent string, target vespa.Target) error {
+func verifyTests(app vespa.ApplicationPackage, target vespa.Target) error {
 	// TODO: system-test, staging-setup and staging-test should be required if the application
 	//       does not have any Java tests.
-	suites := map[string]bool{
-		// suite name: required
-		"system-test":     false,
-		"staging-setup":   false,
-		"staging-test":    false,
-		"production-test": false,
+	suites := []struct {
+		suite    string
+		required bool
+	}{
+		{"system-test", false},
+		{"staging-setup", false},
+		{"staging-test", false},
+		{"production-test", false},
+	}
+	testPath := app.TestPath
+	if app.IsZip() {
+		path, err := app.Unzip(true)
+		if err != nil {
+			return err
+		}
+		defer os.RemoveAll(path)
+		testPath = path
 	}
-	for suite, required := range suites {
```

Comment:
I liked the map approach better as the destructuring here was more clear.

Reply:
agreed, and fixed.